pkg/azureutils/azure_snapshot_utils.go

```diff
@@ -37,7 +37,7 @@ func GetSnapshotAndResourceNameFromSnapshotID(snapshotID string) (snapshotName,
 	return snapshotName, resourceGroup, err
 }
 
-func GenerateCSISnapshot(sourceVolumeID string, snapshot *compute.Snapshot) (*csi.Snapshot, error) {
+func GenerateV1Alpha1Snapshot(sourceVolumeID string, snapshot *compute.Snapshot) (*v1alpha1.Snapshot, error) {
```

Comment:
Consider `NewAzureDiskSnapshot`. We will eventually change the API version used internally: we should avoid using versions in our function names.